consensus/test/member_management_test.go

```diff
@@ -0,0 +1,526 @@
+  checked too, and finally it checks the total stake supply after the membership updates.
+
+  for the other cases in the file: add stake_holder, participants, or remove user, they follow the same work flow and
+  some rules to check the outputs.
+*/
+
+func TestMemberManagementAddNewValidator(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode")
+	}
+	once := sync.Once{}
+	// prepare chain operator
+	operatorKey, err := keygenerator.Next()
+	if err != nil {
+		t.Fatal(err)
+	}
+	operatorAddress := crypto.PubkeyToAddress(operatorKey.PublicKey)
+
+	newValidatorKey, err := keygenerator.Next()
+	require.NoError(t, err)
+
+	stakeBalance := new(big.Int).SetUint64(300)
+
+	addValidatorHook := func(validator *testNode, _ common.Address, _ common.Address) (bool, *types.Transaction, error) { //nolint
+		if validator.lastBlock <= 3 {
+			return true, nil, nil
+		}
+		once.Do(func() {
+			contract, err := autonityInstance(validator.rpcPort)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer contract.Close()
+			txOpt, err := contract.transactionOpts(operatorKey)
+			if err != nil {
+				t.Fatal(err)
+			}
+			eNode := enode.V4DNSUrl(newValidatorKey.PublicKey, "VN:8527", 8527, 8527)
+			_, err = contract.AddValidator(txOpt, crypto.PubkeyToAddress(newValidatorKey.PublicKey), stakeBalance, eNode)
+			if err != nil {
+				t.Fatal(err)
+			}
+		})
+		return false, nil, nil
+	}
+
+	// genesis hook
+	genesisHook := func(g *core.Genesis) *core.Genesis {
+		g.Config.AutonityContractConfig.Operator = operatorAddress
+		g.Alloc[operatorAddress] = core.GenesisAccount{
+			Balance: big.NewInt(100000000000000000),
+		}
+		return g
+	}
+
+	addValidatorCheckerHook := func(t *testing.T, validators map[string]*testNode) {
+		conn, err := ethclient.Dial("http://127.0.0.1:" + strconv.Itoa(validators["VA"].rpcPort))
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer conn.Close()
+
+		contractAddress := autonity.ContractAddress
+		instance, err := NewAutonity(contractAddress, conn)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		auth := bind.CallOpts{
+			Pending:     false,
+			From:        common.Address{},
+			BlockNumber: new(big.Int).SetUint64(validators["VA"].lastBlock),
+			Context:     context.Background(),
+		}
+
+		// check node presented in white list.
+		eNode := enode.V4DNSUrl(newValidatorKey.PublicKey, "VN:8527", 8527, 8527)
+		whiteList, err := instance.GetWhitelist(&auth)
+		if err != nil {
+			t.Fatal(err)
+		}
+		whiteListed := false
+		for _, node := range whiteList {
+			if node == eNode {
+				whiteListed = true
+				break
+			}
+		}
+
+		// check node role and its stake balance.
+		curNetworkMetrics, err := instance.DumpEconomicsMetricData(&auth)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		founded := false
+		for index, v := range curNetworkMetrics.Accounts {
+			if v == crypto.PubkeyToAddress(newValidatorKey.PublicKey) {
+				founded = true
```

Comment:
```suggestion
		found := false
		for index, v := range curNetworkMetrics.Accounts {
			if v == crypto.PubkeyToAddress(newValidatorKey.PublicKey) {
				found = true
```

Found is the past tense of find, founded means something else.